helix-core/src/main/java/org/apache/helix/BaseDataAccessor.java

```diff
@@ -95,6 +95,24 @@
    */
   boolean[] createChildren(List<String> paths, List<T> records, int options);
 
+  /**
+   * Use it when creating children under a parent node with an expected ZK session.
+   * <p>
+   * This will use async api for better performance. If the children already exist it will return
+   * false.
+   *
+   * @param paths the paths to the children ZNodes
+   * @param records List of data to write to each of the path
+   * @param options Set the type of ZNode see the valid values in {@link AccessOption}
+   * @param expectedSession The expected ZK session to create children
+   * @return For each child: true if creation succeeded, false otherwise (e.g. if the child exists)
+   */
+  default boolean[] createChildren(List<String> paths, List<T> records, int options,
```

Comment:
Instead of polluting the interface, shall we have a new interface called sessionAwareZkClient or something like that? Note that an implementation can implement multiple interfaces. So we just need to let the right class implement that interface. And this one won't need to have this ugly and not supported method.

Reply:
Yes. The logic is: sendMessages() -> HelixDataAccessor.createChildren() -> BaseDataAccessor.createChildren() -> ZkClient.asyncCreate().

So we need the expectedSession to be passed all way down to zkclient from sendMessages().

Reply:
I understand your point. By having a new interface, we could add javadoc to explain not use it if they don't care about session (maybe helix internal use only). They still keep using BaseDataAccessor without being confused by the extra session aware api. This is the benefit.

And another benefit is we may add more session aware apis to the new interface in the future if we need.

The downside of having a new interface is 
1. we have to maintain the extra interfaces: SessionAwareBaseDataAccessor, SessionAwareHelixDataAccessor, (maybe session aware zkclient?).
2. Need more code changes to implement the new interfaces (SessionAwareBaseDataAccessor) in this PR.

Let's discuss further.